fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/cursors/SortCursorTests.java

```diff
@@ -0,0 +1,398 @@
+
+import com.apple.foundationdb.record.EndpointType;
+import com.apple.foundationdb.record.ExecuteProperties;
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.RecordCursorResult;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.metadata.expressions.KeyExpression;
+import com.apple.foundationdb.record.provider.common.DynamicMessageRecordSerializer;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTestBase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBStoredRecord;
+import com.apple.foundationdb.record.provider.foundationdb.SortedRecordSerializer;
+import com.apple.foundationdb.record.sorting.FileSortAdapter;
+import com.apple.foundationdb.record.sorting.FileSortCursor;
+import com.apple.foundationdb.record.sorting.MemorySortAdapter;
+import com.apple.foundationdb.record.sorting.MemorySortCursor;
+import com.apple.foundationdb.record.sorting.MemorySorter;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.Tags;
+import com.google.protobuf.CodedInputStream;
+import com.google.protobuf.CodedOutputStream;
+import com.google.protobuf.Message;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.io.File;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests for {@link MemorySortCursor} and {@link FileSortCursor}.
+ */
+@Tag(Tags.RequiresFDB)
+public class SortCursorTests extends FDBRecordStoreTestBase {
+    final KeyExpression num2Field = Key.Expressions.field("num_value_2");
+
+    private SortedRecordSerializer<Message> serializer;
+
+    private List<Integer> sortedNums;
+
+    @BeforeEach
+    public void setupRecords() throws Exception {
+        sortedNums = new ArrayList<>(100);
+        Random random = new Random(3456);
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+
+            serializer = new SortedRecordSerializer<>(DynamicMessageRecordSerializer.instance(), recordStore.getRecordMetaData(), recordStore.getTimer());
+
+            for (int i = 0; i < 100; i++) {
+                int num = random.nextInt();
+                sortedNums.add(num);
+
+                TestRecords1Proto.MySimpleRecord.Builder record = TestRecords1Proto.MySimpleRecord.newBuilder();
```

Comment:
does anything in these cursors actually depend on transactions or record stores? Some integration tests are valuable, but I think the majority of the behavior could be tested with just a `ListCursor` underneath.